File: 프로그래머스/1/159994. 카드 뭉치/카드 뭉치.py
def solution(cards1, cards2, goal):
    idx = 0
    while idx < len(goal):
        if cards1 and cards1[0] == goal[idx]:
            cards1.pop(0)
            idx += 1
        elif cards2 and cards2[0] == goal[idx]:
            cards2.pop(0)
            idx += 1
        else:
            break
    return 'Yes' if idx == len(goal) else 'No'

File: 프로그래머스/1/159994. 카드 뭉치/카드 뭉치.java
class Solution {
    public String solution(String[] cards1, String[] cards2, String[] goal) {
        int idx = 0;
        int cards1Idx = 0; int cards2Idx = 0;
        while (idx < goal.length) {
            if (cards1Idx < cards1.length && cards1[cards1Idx].equals(goal[idx])) {
                cards1Idx += 1;
                idx += 1;
            }
            else if (cards2Idx < cards2.length && cards2[cards2Idx].equals(goal[idx])) {
                cards2Idx += 1;
                idx += 1;
            } else {
                break;
            }
        }
        return (idx == goal.length) ? "Yes" : "No";
    }
}